CDCgov/prime-reportstream
# Add hospital receiver to Flexion org settings for results flow

This PR adds the `simulated-hospital` receiver settings for the TI results use case as described in [TI's issue #641](https://github.com/CDCgov/trusted-intermediary/issues/641), along with test ORU files for the translation test. It also fixes the looping issue with TI by filtering using the `ETOR` meta tag and adds the routing filter for the receivers

## Local Tests

### Steps to test Results:
1. Run `./prime multiple-settings set --input ./settings/staging/0166-flexion-staging-results-handling.yml`
2. Choose an ORU message to send to `flexion.simulated-lab`. You may use `prime-router/src/testIntegration/resources/datatests/HL7_to_FHIR/etor_ORU_20240220.hl7` as a sample ORU
4. Change the Receiving Facility ID (`MSH-6.2`) in the ORM to `simulated-hsopital-id`
5. Send the ORU to the waters endpoint:
    `curl --header 'Content-Type: application/hl7-v2' --header 'Client: flexion.simulated-lab' --header 'Authorization: Bearer token' --data-binary '@/path/to/etor_ORU_20240220.hl7' 'http://localhost:7071/api/waters'`
4. Verify the request went through without any errors and uploaded to the SFTP for the hospital. The delivered `ORU` message should map everything in the original `ORU` with the exception of `OBX.4` and `OBX.29`

### Steps to test Orders:
1. Make sure `./settings/staging/0166-flexion-staging-results-handling.yml` has been applied locally
2. Choose an ORM message to send to `flexion.simulated-hospital`. You may use `prime-router/src/testIntegration/resources/datatests/HL7_to_FHIR/sample_orm_20230809-001.hl7` as a sample ORM
4. Change the Receiving Facility ID (`MSH-6.2`) in the ORM to `simulated-lab-id`
5. Send the ORM to the waters endpoint:
    `curl --header 'Content-Type: application/hl7-v2' --header 'Client: flexion.simulated-hospital' --header 'Authorization: Bearer token' --data-binary '@/path/to/sample_orm_20230809-001.hl7' 'http://localhost:7071/api/waters'`
6. Verify the request went through without any errors and uploaded to the SFTP for the lab. The message should be delivered only once

### Steps to test final destination routing
1. Change `MSH-6.2` in the message to anything other than what's instructed above and send the message
2. The message should not be delivered to its final destination

### Looping issue
If you send a FHIR message as `etor-service-sender` without the `ETOR` meta tag, the message should be sent back to TI instead of routed to deliver to destination. You could use this [ORU FHIR message](https://github.com/CDCgov/trusted-intermediary/blob/main/examples/MN/004_MN_ORU_R01_NBS_1_hl7_translation.fhir) that doesn't have the ETOR tag 

## Checklist

### Testing
- [x] Ran `./gradlew testSmoke`
- [x] Ran `./gradlew testIntegration`
